app/src/lib/git/stage.ts

```diff
@@ -54,15 +54,33 @@ export async function stageManualConflictResolution(
       )).exitCode
       break
     }
-    case GitStatusEntry.Added:
-    case GitStatusEntry.UpdatedButUnmerged: {
+    case GitStatusEntry.Added: {
       exitCode = (await git(
         ['add', file.path],
         repository.path,
         'addConflictedFile'
       )).exitCode
       break
     }
+    case GitStatusEntry.UpdatedButUnmerged: {
+      const choiceFlag =
+        manualResolution === ManualConflictResolutionKind.theirs
+          ? 'theirs'
+          : 'ours'
+      exitCode = (await git(
+        ['checkout', `--${choiceFlag}`, '--', file.path],
+        repository.path,
+        'checkoutConflictedFile'
+      )).exitCode
+      if (exitCode === 0) {
+        exitCode = (await git(
+          ['add', file.path],
+          repository.path,
+          'addConflictedFile'
+        )).exitCode
```

Comment:
:art: not a blocker, but seeing how we poke at the `exitCode` within a switch statement a few times I wonder if there's a better way to organize this logic.


 -   introduce a simple wrapper to de-duplicate the same work that each Git operation does in this function:

```ts
/**
 * Run a Git command and return whether the exit code indicated success
 *
 * This defers to the default error handling infrastructure inside if an error
 * is encountered.
 */
async function runGitCommand(
  args: string[],
  path: string,
  name: string
): Promise<boolean> {
  const { exitCode } = await git(args, path, name)
  return exitCode === 0
}
```

 - tidy up the simple case to make it a bit more readable:

```ts
    case GitStatusEntry.Deleted: {
      return await runGitCommand(
        ['rm', file.path],
        repository.path,
        'removeConflictedFile'
      )
    }
```

 - For this new case, we also get a bit more readability:

```
      const choiceFlag =
        manualResolution === ManualConflictResolutionKind.theirs
          ? 'theirs'
          : 'ours'
      const checkoutCompleted = await runGitCommand(
        ['checkout', `--${choiceFlag}`, '--', file.path],
        repository.path,
        'checkoutConflictedFile'
      )
      if (checkoutCompleted) {
        return await runGitCommand(
          ['add', file.path],
          repository.path,
          'addConflictedFile'
        )
      }
      return false
```

This should make the variable `exitCode` unnecessary, and you can `return assertNever` below to make the compiler happy.